controllers/webhooksController.js

```diff
@@ -24,117 +27,121 @@ module.exports = (() => {
+          invoiceSucceededCase(event, invoiceMap);
           break;
-        case 'invoice.payment_failed':
-          session = event.data.object;
-          const endDate = moment.utc(session.lines.data[0].period.end * 1000).toString();
-          const startDate = moment.utc(session.lines.data[0].period.start * 1000).toString();
-          try {
-            const subscription = await subsRepo.getSubscriptionByCustomerID(session.customer);
-            const plan = await plansRepo.getPlanByStripeId(session.lines.data[0].price.product);
-            const jsonObj = {
-              accountId: subscription.accountId,
-              plan: plan._id.toString(),
-              start_date: startDate,
-              next_date: endDate,
-              renewal: endDate,
-              status: 'suspended',
-              customerId: session.customer,
-              stripeSubId: session.subscription
-            };
-            await subsRepo.editSubscription(subscription._id.toString(), jsonObj);
-            sendSuspendedAccountToIAM(subscription.accountId);
-            logger.info(`${subscription.accountId}'s details sent to IAM team.`);
-            logger.info(`${subscription.accountId} subscription was suspended.`);
-          } catch (err) {
-            logger.error(`failed to Update the repository: ${err.message}`);
-            res.status(404).send(err.message);
-          }
+        case invoiceFailed:
+          invoiceFailedCase(event);
           break;
-        case 'customer.subscription.deleted':
-          session = event.data.object;
-          const date = moment.utc(session.ended_at * 1000).toString();
-          const nextDate = new Date();
-          nextDate.setDate(nextDate.getDate() + 1);
-          try {
-            const subscription = await subsRepo.getSubscriptionByCustomerID(session.customer);
-            const plan = await plansRepo.getPlanByName('Free');
-            const jsonObj = {
-              plan: plan._id.toString(),
-              start_date: date,
-              next_date: nextDate,
-              renewal: date,
-              status: 'active',
-              customerId: session.customer,
-              stripeSubId: ''
-            };
-            await subsRepo.editSubscription(subscription._id.toString(), jsonObj);
-            logger.info(`${subscription.accountId} subscription was updated.`);
-            sendSubscriptionToIAM(subscription.accountId, plan.credits, plan.seats, plan.features);
-            logger.info(`${subscription.accountId}'s details sent to IAM team.`);
-          } catch (err) {
-            logger.error(`failed to Update the repository: ${err.message}`);
-            res.status(404).send(err.message);
-          }
+        case customerDeleted:
+          customerDeletedCase(event);
           break;
       }
       res.status(200).send();
     }
   });
 })();
+
+const invoiceSucceededCase = async (event, invoiceMap) => {
+  const session = event.data.object;
```

Comment:
I see that in all the cases you do this "session = event.data.object"
You can maybe send "event.data.object" to the function and write "session" as one of the parameters

it will look nicer I think